crates/oxc_linter/src/rules/typescript/consistent_type_definitions.rs

```diff
@@ -0,0 +1,214 @@
+use oxc_ast::{
+    ast::{ExportDefaultDeclarationKind, TSType},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("typescript-eslint(consistent-type-definitions):")]
+#[diagnostic(severity(warning), help("Use an `{0}` instead of a `{1}`"))]
+struct ConsistentTypeDefinitionsDiagnostic(
+    &'static str,
+    &'static str,
+    #[label("Use an `{0}` instead of a `{1}`")] pub Span,
+);
+
+#[derive(Debug, Default, Clone)]
+pub struct ConsistentTypeDefinitions {
+    config: ConsistentTypeDefinitionsConfig,
+}
+
+#[derive(Debug, Default, Clone, Copy, Eq, PartialEq)]
+enum ConsistentTypeDefinitionsConfig {
+    #[default]
+    Interface,
+    Type,
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    ///
+    /// ### Why is this bad?
+    ///
+    ///
+    /// ### Example
+    /// ```javascript
+    /// ```
+    ConsistentTypeDefinitions,
+    correctness
```

Comment:
This is a `style` rule from the docs https://typescript-eslint.io/rules/consistent-type-definitions/

Reply:
my bad, fixed